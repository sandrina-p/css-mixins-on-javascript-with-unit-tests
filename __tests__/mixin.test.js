const mixins = require('./../src/styles/mixins');

const mock = {
    replaceWith: jest.fn(),
};

test('button default generates with correct styles', () => {
    expect(mixins.button(mock, 'teal', 'white', 'tomato')).toMatchSnapshot();
});

test('padding generates with small size', () => {
    expect(mixins.padding(mock, 'sm')).toMatchSnapshot();
});

test('font-size render with 20px - 1.25rem', () => {
    expect(mixins.fontSize(mock, '20')).toMatchSnapshot();
});
