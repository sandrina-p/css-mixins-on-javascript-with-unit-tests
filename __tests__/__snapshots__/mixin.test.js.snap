exports[`test button default generates with correct styles 1`] = `
Object {
  "&:hover, &:focus": Object {
    "background": "tomato",
  },
  "background": "teal",
  "color": "white",
  "padding": "0.5rem",
}
`;

exports[`test font-size render with 20px - 1.25rem 1`] = `
Object {
  "@media (min-width: 60em)": Object {
    "font-size": "1.375rem",
  },
  "font-size": "1.25rem",
}
`;

exports[`test padding generates with small size 1`] = `
Object {
  "padding": "0.2rem 0.5rem",
}
`;
